src/averageAges.js

```diff
@@ -1,64 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const totalAge = (array) => {
+  return array.reduce(
+    (acc, per) => acc + (per.died - per.born), 0);
+};
+
+const conditionCheck = (firstCondition, secondCondition) => {
+  return firstCondition === secondCondition;
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const newPeople = century ? people.filter(
+    person => conditionCheck(person.sex, 'm')
+    && Math.ceil(person.died / 100) === century)
+    : people.filter(person => conditionCheck(person.sex, 'm'));
+
+  return totalAge(newPeople) / newPeople.length;
```

Comment:
let's make your code cleaner
```suggestion
  const newPeople = century ? people.filter(person =>
    conditionCheck(person.sex, 'm') &&
    Math.ceil(person.died / 100) === century
   )
   : people.filter(person => conditionCheck(person.sex, 'm'));

  return totalAge(newPeople) / newPeople.length;
```